src/move_basic.cpp

```diff
@@ -492,18 +492,19 @@ void MoveBasic::executeAction(const move_base_msgs::MoveBaseGoalConstPtr& msg)
             return;
         }
         sleep(localizationLatency);
+    }
 
-        // Final rotation as specified in goal
-        if (do_final_rotation) {
-            double finalYaw = goalYaw - (yaw + requestedYaw);
-            if (std::abs(finalYaw) > angularTolerance) {
-                if (!rotate(finalYaw, drivingFrame)) {
-                    return;
-                }
-            }
-
-            sleep(localizationLatency);
+    // Final rotation as specified in goal
+    if (do_final_rotation) {
+        double requestedYaw = atan2(linear.y(), linear.x());
+        double finalYaw = goalYaw - (yaw + requestedYaw);
+        if (std::abs(finalYaw) > angularTolerance) {
+	    if (!rotate(finalYaw, drivingFrame)) {
+	        return;
+	    }
         }
+
+        sleep(localizationLatency);
     }
 
     actionServer->setSucceeded();
```

Comment:
@MoffKalast this line is setting it as succeeded

Reply:
It doesn't seem to get that far apparently. I'm not sure what rotate() gives back at specific times, but I'd imagine it just returns without setting the status?

Reply:
Hmm, yes it could be possible that the boolean we are returning from rotate function is wrong. If you are able to reproduce the issue, reopen it.